### PR TITLE
Create views fixes

### PR DESCRIPTION
**Pull request by cymed**
This PR adds the following functionalities:

- [x] As mentioned in #147, we need to separate tww_app rights from all the other rights as we intend to drop and re-create the schema on update
- [x] New script in datamodel package to drop and re-create tww_app
- [x] makes create_views callable from outside the docker environment by relying on abspath
- [x] adds 6 singleinheritance views that for some reason remained in the dumps but were not created by create_views.py
- [x] fixes vw_tww_import to directly access tww_od schemas instead of tww_app.vw_tww_wastewater_structure

### CODE REVIEW

**Comment by cymed** [comment]
in Unit Tests:
/src/datamodel/scripts/setup.sh: line 36: /src/datamodel/scripts/../app/create_tww_app.py: Permission denied
Any Ideas why?

**Comment by 3nids** [comment]
@cymed `chmod +x create_tww_app.py`

File: datamodel/scripts/setup.sh [hunk 1]
```diff
@@ -32,5 +32,5 @@ psql "service=${PGSERVICE}" -v ON_ERROR_STOP=1 -f ${DIR}/changelogs/0001/52_plan
 psql "service=${PGSERVICE}" -v ON_ERROR_STOP=1 -f ${DIR}/12_0_roles.sql
 psql "service=${PGSERVICE}" -v ON_ERROR_STOP=1 -f ${DIR}/12_1_roles.sql
 
-
+chmod +x ${DIR}/app/create_tww_app.py
```
Comment (3nids) [inline datamodel/scripts/setup.sh:35]: no, this needs to be done on your system (make the file executable)

Comment (cymed) [reply datamodel/scripts/setup.sh:35]: Which is a bit problematic when you work on windows and don't have the necessary rights to install git on your shell

Comment (cymed) [reply datamodel/scripts/setup.sh]: OK I managed that part. However, setup.sh does not treat create_tww_app as a python script anymore:

2024-02-09T06:54:04.6558778Z /src/datamodel/scripts/../app/create_tww_app.py: line 1: import: command not found
2024-02-09T06:54:04.6559590Z /src/datamodel/scripts/../app/create_tww_app.py: line 2: import: command not found
2024-02-09T06:54:04.6560370Z /src/datamodel/scripts/../app/create_tww_app.py: line 4: import: command not found
2024-02-09T06:54:04.6561163Z /src/datamodel/scripts/../app/create_tww_app.py: line 5: from: command not found
2024-02-09T06:54:04.6562024Z /src/datamodel/scripts/../app/create_tww_app.py: line 8: syntax error near unexpected token `('
2024-02-09T06:54:04.6563144Z /src/datamodel/scripts/../app/create_tww_app.py: line 8: `def run_sql_file(file_path: str, pg_service: str, variables: dict = {}):'

Comment (3nids) [reply datamodel/scripts/setup.sh]: you need to add `#!/usr/bin/env python` on top

Comment (cymed) [reply datamodel/scripts/setup.sh]: as in https://github.com/teksi/wastewater/pull/149/commits/875eb1d088e82f38d0378a2c51643b7ffb1792f4? Or do we need python3?

**Comment by cymed** [comment]
The only thing that is not working here is the shebang. Can someone with linux experience set up the shell script please?

**Review by ponceta — COMMENTED** [review]
add shebang

**Review by ponceta — APPROVED** [review]
Main fixes and prepare work for release and app re-creation.

@3nids if you can give it a look so we don't introduce anything bad for the release management process (future)

**Comment by 3nids** [comment]
One more thought on the way to pass variables.

There is another thing to consider: moving to psycopg3 from psycopg2. It is quite a minor change, and probably easier to do at early stage of the project (before an official release).

Another approach is recommended in the doc:https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#multiple-statements-in-the-same-query (using `sql.SQL.format`)

So, I am not sure what is the most reasonable approach here, but the best would be integrate the move to psycopg3 into this PR. Not sure if @ponceta  agrees and if that's possible on your end @cymed. I can help if needed.

**Comment by ponceta** [comment]
@3nids is it possible to upgrade pirogue to psycopg3 as well?

**Comment by 3nids** [comment]
@ponceta yes pirogue is an easy one to upgrade.

**Comment by 3nids** [comment]
superseded by #197

**Pull request closed without merging**